### PR TITLE
'Query' support for channels.list.joined, groups.list, groups.listAll, im.list

### PR DESCRIPTION
**Pull request by xbolshe**

### CODE REVIEW

File: developer-guides/rest-api/groups/listAll/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Group List All
```
Comment (MartinSchoeler) [inline developer-guides/rest-api/groups/listAll/README.md:1]: Please add this file to the [Table of Contents](https://github.com/RocketChat/docs/blob/master/_data/toc.yml)

**Review by MartinSchoeler — DISMISSED** [review]

File: developer-guides/rest-api/groups/listAll/README.md (hunk 1 above)
Comment (xbolshe) [reply developer-guides/rest-api/groups/listAll/README.md:1]: Added.

File: developer-guides/rest-api/im/list/README.md [hunk 2]
```diff
@@ -56,11 +56,61 @@ curl -H "X-Auth-Token: 9HqLlyZOugoStsXCUfD_0YdwnNnunAJF8V47U3QHXSq" \
     ],
     "success": true
```
Comment (MarcosSpessatto) [inline developer-guides/rest-api/im/list/README.md]: Hey @xbolshe, can you add the `offset`, `count` and `total` properties to this response object please?

Comment (xbolshe) [reply developer-guides/rest-api/im/list/README.md]: It seems `"t": "p"` is incorrect for Direct Messages. I updated request/response using real Rocket Chat.

**Review by MarcosSpessatto — APPROVED** [review]

**Pull request merged**